### PR TITLE
fix(rules): deserialize optional filters

### PR DESCRIPTION
**Pull request by aallam**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no    
| BC breaks?        | no     
| Related Issue     | n/a
| Need Doc update   | no

## Describe your change

Handle `optionalFilters` deserialization in the case of mixed formats (String and List)

### CODE REVIEW

File: algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java [hunk 1]
```diff
@@ -212,6 +212,16 @@ void testLegacyFiltersFormat(String input) throws IOException {
             Defaults.getObjectMapper().readValue(nestedArrayFilters, ConsequenceParams.class),
             input));
 
+    // Testing mixed case with array and string
+    // [["color:green","color:yellow"],"color:blue"]
+    String stringAndArrayFilters =
```
Comment (Ant-hem) [inline algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java:217]: We have the same issue with the C# client and Go, we probably not handle this use-case.

**Review by Ant-hem — APPROVED** [review]
Thanks for the fix 👍🏻

File: algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java (hunk 1 above)
Comment (aallam) [reply algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java:217]: True, I've checked the C# client, didn't see something related to this edge case
